### ISSUE TITLE
PatchSystem.sh script buggy?!

### ISSUE THREAD

**Issue by shines75** [Ursinia/Mini-Monterey-Patcher#11]
Hey, Ben Sova! What the heck is this? I can understand that your PatchSystem.sh shell script is still in alpha development, but is completely buggy! If run from the Monterey Install disk, the script exits at line 54, for “a command not found”, and, on the other hand, if run from my big sur installation, (full macos mode), gives a plenty of errors, too many to list, that personally I think it’s shameful!!
I’m so sorry to say that, because you’re a talented hacker and developer, but…too bad this time, Ben Sova.

**Comment by ozziepadilla** [comment]
What version of mini Monterey patcher are you using ? Make sure it’s v0.0.5b because right now I am doing the usb and I didn’t get any error and please show more respect to Ben he isn’t a hacker , nobody has done what Ben it’s doing for everyone to give more life time to the old MacBooks and iMac others patchers wires but personally ben’s patcher are the best period

**Comment by BenSova** [comment]
Can you send me the full error?

**Comment by BenSova** [comment]
Also how did you get here if PatchUSB.sh wasn't working?

And one other thing, can you send me a picture of the contents of your USB drive?

**Comment by shines75** [comment]
Forget it all. I do apologise, but I downloaded the Mini-Monterey-Patcher-main and not Mini-Monterey-Patcher-0.0.5b!!! I've been such a bummer...I'll retry with the correct version, and now everything should go fine.

**Comment by shines75** [comment]
But wait! According to the readme that comes with version 0.0.5b, for my MacBook Pro mid-2012 there's no 802.11n patching, so...what's wrong? With the main version that should be fixed, doesn't it? And now?

**Comment by shines75** [comment]
Do I have to wait a little yet?

**Comment by ozziepadilla** [comment]
> But wait! According to the readme that comes with version 0.0.5b, for my MacBook Pro mid-2012 there's no 802.11n patching, so...what's wrong? With the main version that should be fixed, doesn't it? And now?

A new release it’s on the way but not yet that allow to get wifi working on MacBook like yours and others models too si your MacBook won’t have wifi working for now , gives some time

**Comment by ozziepadilla** [comment]
> Do I have to wait a little yet?

That’s correct

**Comment by shines75** [comment]
I've to say I'm pretty new to the Github's world...let me understand, please: what is exactly the difference between the main version of a software, and its latest release? Moreover, It is correct to say that the main version of Mini-Monterey-Patcher has 802.11n patching already fixed on my Mac?

**Comment by ozziepadilla** [comment]
> I've to say I'm pretty new to the Github's world...let me understand, please: what is exactly the difference between the main version of a software, and its latest release? Moreover, It is correct to say that the main version of Mini-Monterey-Patcher has 802.11n patching already fixed on my Mac?

Again ,,,, not yet ,,, you must wait for the next release that will Patchet your MacBook wifi but  still under development not need to rush by the way macOS Monterey its still in beta and there you can say it’s very buggy but not the mini Monterey patcher capiche???

**Comment by shines75** [comment]
Okay, okay, I already understood this...but, please, could you explain to a novice of Github, what exactly is the difference between the main and the latest version of a software? Let's make it clear!

**Comment by Monkiey** [comment]
> Okay, okay, I already understood this...but, please, could you explain to a novice of Github, what exactly is the difference between the main and the latest version of a software? Let's make it clear!

the main branch is where the dev put all of the code that is in active development, the release is the latest version of the software(code from the main branch )that is ironed out and ready for prime time and packaged up as such, hope that clears it up fo you

**Comment by shines75** [comment]
Aaah, interesting...so we can say pre802.11ac wifi patching is in the works for Mini-Monterey-Patcher!!

**Comment by Monkiey** [comment]
> Aaah, interesting...so we can say pre802.11ac wifi patching is in the works for Mini-Monterey-Patcher!!

Exactly you can see in the comments ben has put code that is active development for the wifi patching

**Comment by shines75** [comment]
Eh, reassuring!! Thus, there's only to wait...but I suppose that for you, devs, now it's impossible to say "when", right?

**Comment by Monkiey** [comment]
> Eh, reassuring!! Thus, there's only to wait...but I suppose that for you, devs, now it's impossible to say "when", right?

Yeah, not sure if I can answer that as I am not the dev of this project you would have to ask ben about that.

**Comment by shines75** [comment]
Naaa, I'm such a pain in the ass...I really think now I've to drop. Thank you  very much. Bye

**Comment by Monkiey** [comment]
> Naaa, I'm such a pain in the ass...I really think now I've to drop. Thank you very much. Bye

no problem, have a great day

**Comment by BenSova** [comment]
I'll have it later today hopefully.

**Comment by shines75** [comment]
Unfortunately, even with the latest version of the patcher, 0.0.5b, the question is still open: I get stuck with the same errors of the main version tried before...both in monterey installer mode and in full macos mode! What's wrong, now?

**Comment by Monkiey** [comment]
> Unfortunately, even with the latest version of the patcher, 0.0.5b, the question is still open: I get stuck with the same errors of the main version tried before...both in monterey installer mode and in full macos mode! What's wrong, now?

can you post screenshots of the errors please

**Comment by shines75** [comment]
Yes, wait 5 mins

**Comment by shines75** [comment]
![Full macOS Mode](https://user-images.githubusercontent.com/87176050/125131932-e51d8280-e103-11eb-85bf-0f6bc579d3c4.jpg)
![Recovery Mode](https://user-images.githubusercontent.com/87176050/125131940-e77fdc80-e103-11eb-81b7-008d9c4106c0.jpg)

**Comment by Monkiey** [comment]
Yeah, I am not sure how to help i just ran the patch system for big sur the other day and it worked fine not Mini Monterey tho

**Comment by ozziepadilla** [comment]
First you have to download the InstallAssistant.pkg then install it and that will be installed in your applications Folder make sure you are typing in Terminal like this sudo /Applications/Install\ macOS\ Monterey\ beta.app/Contents/Resources/createinstallmedia --volume     make sure there is a space after createinstallmedia and after --volume then draw in your Usb and press enter, then type in your password

**Comment by shines75** [comment]
Nope. I recreated the Monterey USB Install disk with the createinstallmedia tool from Apple correctly

**Comment by shines75** [comment]
The issue is inside the script made by Ben Sova

**Comment by shines75** [comment]
I think It has to be deeply rewritten

**Comment by shines75** [comment]
Well guys, if you don't know how to help me, I think I'll have to pass the ball to Ben

**Comment by Monkiey** [comment]
> First you have to download the InstallAssistant.pkg then install it and that will be installed in your applications Folder make sure you are typing in Terminal like this sudo /Applications/Install\ macOS\ Monterey\ beta.app/Contents/Resources/createinstallmedia --volume make sure there is a space after createinstallmedia and after --volume then drag in your Usb and press enter, then type in your password

no this script is different this the one to patch the kexts after install not patch the USB

**Comment by BenSova** [comment]
First Error: You can't patch Monterey from Big Sur, I probably need to include a check for that.

Second Error: I thought I fixed that. I don't think the script didn't save (that's happened so many times with this project, it's crazy at this point.)

**Comment by BenSova** [comment]
I'll fix both real quick.

**Comment by shines75** [comment]
Hi, Ben! So, Are you telling me there're errors in your PatchSystem.sh script?

**Comment by Monkiey** [comment]
> Hi, Ben! So, Are you telling me there're errors in your PatchSystem.sh script?

yes, we are all human and we make errors sometimes ben has aid he will fix them asap

**Comment by shines75** [comment]
Yes, I understand...one more thing, a question I'd make directly to Ben: when will you add pre802.11ac support for Macs prior to mid-2013? Answer me, please!

**Comment by Monkiey** [comment]
> Yes, I understand...one more thing, a question I'd make directly to Ben: when will you add pre802.11ac support for Macs prior to mid-2013? Answer me, please!

Ben has said earlier in the thread that it will be today at the latest

**Comment by BenSova** [comment]
In the next version

Btw cf1565d fixes your errors, so just wait until the release later today and you'll have the pre-802.11ac stuff and a fix to your errors.

**Comment by BenSova** [comment]
Btw I responded to your other issue (#10), could you respond back?

**Comment by shines75** [comment]
Fantastic, good work then! Bye!!

**Comment by GitHub3590** [comment]
Hi Ben I posted somewhere here that I just tried the pre-802.11ac patcher that just came out about an hour ago.. I booted into the USB and finally typed everything correctly and the patcher finished successfully but when I boot back into Monterey I get no wifi... It don't show up anywhere. I'm using a Mac Pro 5,1 with Supported PHY Modes:	802.11 a/b/g/n.
Am i doing something wrong?  It said it was patched successfully..
Thanks in advance!
